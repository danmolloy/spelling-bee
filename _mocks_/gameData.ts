import { faker } from "@faker-js/faker"

//const centerLetter = faker.string.alpha();
//const outerLetters = faker.string.alpha(6).split("")
const expiration = Date.now() + 10000
const answers = faker.word.words(25).toLowerCase().split("").filter(i => i !== " ")
const lettersArr = Array.from(new Set(answers)).slice(0, 7)

const validLetters = Array.from(new Set(faker.string.alpha(12).split(""))).slice(0, 7)

export const mockData = {
  expiration: expiration,
  displayWeekday:"Sunday",
  displayDate: "October 13, 1986",
  printDate:"2022-04-10",
  centerLetter: lettersArr[0],
  outerLetters: lettersArr.slice(1, 7),
  validLetters: lettersArr,
  pangrams: [lettersArr.join("")],
  answers: [lettersArr.join(""), lettersArr.slice(1, 6).join(""), ...faker.word.words(25).split(" ")],
  id: faker.string.uuid(),
  freeExpiration:"0",
  editor:"Sam Ezersky"
}