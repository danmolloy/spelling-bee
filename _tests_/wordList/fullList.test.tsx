import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"
import FullList, { FullListProps } from "../../components/wordList/fullList";
import { mockData } from "../../_mocks_/gameData";



describe("<FullList />", () => {
  const mockProps: FullListProps =  {
  words: ["dfw", "weow"],
  revealWords: false,
  answers: mockData.answers,
  pangrams: mockData.pangrams
}
  beforeEach(() => {
    render(<FullList {...mockProps} />)
  })
  it("full-list-div is in the document", () => {
    const fullList = screen.getByTestId("full-list-div")
    expect(fullList).toBeInTheDocument()
  })
  it("all found words are in the document", () => {
    for (let i = 0; i < mockProps.words.length; i ++) {
      const capitalizedWord = mockProps.words[i][0].toUpperCase() + mockProps.words[i].slice(1)
      const word = screen.getByText(capitalizedWord)
      expect(word).toBeInTheDocument()
    }
  })
  it("no answers are shown if !revealWords", () => {
    const fullList = screen.getByTestId("full-list-div")
    for (let i = 0; i < mockProps.answers.length; i ++) {
      const capitalizedWord = mockProps.answers[i][0].toUpperCase() + mockProps.answers[i].slice(1)
      expect(fullList.textContent).not.toMatch(capitalizedWord)
    }
  })
})

describe("<FullList />", () => {
  const mockProps: FullListProps =  {
    words: [],
    revealWords: true,
    answers: mockData.answers,
    pangrams: mockData.pangrams
  }
  beforeEach(() => {
    render(<FullList {...mockProps} />)
  })
  it("all answers are shown if revealWords === true", () => {
    const fullList = screen.getByTestId("full-list-div")
    for (let i = 0; i < mockProps.answers.length; i ++) {
      const capitalizedWord = mockProps.answers[i][0].toUpperCase() + mockProps.answers[i].slice(1)
      expect(fullList.textContent).toMatch(capitalizedWord)
    }
  })
})

describe("<FullList />", () => {
  const mockProps: FullListProps =  {
    words: [],
    revealWords: true,
    answers: ["lorem", "ipsum"],
    pangrams: []
  }
  beforeEach(() => {
    render(<FullList {...mockProps} />)
  })
  it("matches snapshot", () => {
    const fullList = screen.getByTestId("full-list-div")
    expect(fullList).toMatchSnapshot()
  })
}) 