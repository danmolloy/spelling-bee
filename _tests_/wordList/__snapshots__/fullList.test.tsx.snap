// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullList /> matches snapshot 1`] = `
<div
  class="w-full"
  data-testid="full-list-div"
>
  <div
    class="w-full flex flex-col flex-wrap "
  />
</div>
`;
